pkg/daemon/ceph/cleanup/disk.go

```diff
@@ -174,25 +191,107 @@ func (s *DiskSanitizer) buildShredArgs(disk string) []string {
 	return shredArgs
 }
 
+func (s *DiskSanitizer) buildQuickShredCommands(disk, dataSource string) []ShredCommand {
+	var quickShredCommands []ShredCommand
+
+	diskInfo, err := clusterd.PopulateDeviceInfo(disk, s.context.Executor)
+	if err != nil {
+		return nil
+	}
+
+	// Shred more data at offset 0
+	quickShredCommands = append(quickShredCommands, ShredCommand{command: quickShredUtility, args: []string{
+		fmt.Sprintf("if=%s", dataSource),
+		fmt.Sprintf("of=%s", disk),
+		fmt.Sprintf("bs=%d", quickShredInitialBS),
+		fmt.Sprintf("count=%d", quickShredInitialBlockCount),
+		"oflag=direct,dsync",
+		"seek=0",
+	}})
+
+	// Shred at offsets 1GB, 10GB, 100GB, 1000GB
+	for _, offset := range quickShredOffsets {
+		shredUntil := offset + quickShredOffsetsBS*quickShredOffsetsBlockCount
+
+		// Break if disk size is less than offset + shred size
+		if shredUntil > diskInfo.Size {
+			break
+		}
+
+		quickShredCommands = append(quickShredCommands, ShredCommand{command: quickShredUtility, args: []string{
+			fmt.Sprintf("if=%s", dataSource),
+			fmt.Sprintf("of=%s", disk),
+			fmt.Sprintf("bs=%d", quickShredOffsetsBS),
+			fmt.Sprintf("count=%d", quickShredOffsetsBlockCount),
+			"oflag=direct,dsync,seek_bytes",
+			fmt.Sprintf("seek=%d", offset),
+		}})
+	}
+
+	return quickShredCommands
+}
+
+func (s *DiskSanitizer) buildShredCommands(disk string) []ShredCommand {
+	var shredCommands []ShredCommand
+
+	if s.sanitizeDisksSpec.Method == cephv1.SanitizeMethodQuick {
+		if s.sanitizeDisksSpec.DataSource == cephv1.SanitizeDataSourceRandom {
+			shredCommands = append(shredCommands, s.buildQuickShredCommands(disk, "/dev/urandom")...)
+		}
+
+		shredCommands = append(shredCommands, s.buildQuickShredCommands(disk, "/dev/zero")...)
+
+		return shredCommands
+	}
+
+	if s.sanitizeDisksSpec.DataSource == cephv1.SanitizeDataSourceZero {
+		shredCommands = append(shredCommands, ShredCommand{command: completeShredUtility, args: s.buildShredArgs(disk)})
+		return shredCommands
+	}
+
+	shredCommands = append(shredCommands, ShredCommand{command: completeShredUtility, args: s.buildShredArgs(disk)})
+
+	return shredCommands
+}
+
 func (s *DiskSanitizer) executeSanitizeCommand(osdInfo oposd.OSDInfo, wg *sync.WaitGroup) {
 	// On return, notify the WaitGroup that we’re done
 	defer wg.Done()
 
-	for _, device := range []string{osdInfo.BlockPath, osdInfo.MetadataPath, osdInfo.WalPath} {
+	for _, device := range []string{osdInfo.BlockPath /*, osdInfo.MetadataPath, osdInfo.WalPath*/} {
 		if device == "" {
 			continue
 		}
 
-		output, err := s.context.Executor.ExecuteCommandWithCombinedOutput(shredUtility, s.buildShredArgs(device)...)
+		// output, err := s.context.Executor.ExecuteCommandWithCombinedOutput("ceph-volume", "lvm", "zap", "--destroy", "--osd-id", strconv.Itoa(osdInfo.ID))
```

Comment:
Per discussion, let's go ahead and execute this ceph-volume command, no need for complication of a separate image.

Reply:
Sounds good, but I would need some guidance on encrypted PVC drives as the test is currently failing, it is reporting PVC name as empty, so maybe something specific with the test segtup?

Also do we need to execute this command or would it make sense to reuse the DestroyOSD function?

Reply:
Oh, I overlooked that `ceph-volume lvm zap` is already called in `DestroyOSD()` [here](https://github.com/rook/rook/pull/15666/files#diff-2db24d410e5bc812291568a7c4d775d39f7b99821ec94f132748ad9d90cd6995R286). I wonder why that isn't already wiping the disk as expected? Maybe I'm missing some difference in the args.